pkg/apis/pravega/v1beta1/pravega.go

```diff
@@ -323,6 +323,21 @@ func (s *PravegaSpec) withDefaults() (changed bool) {
 		}
 	}
 
+	if s.SegmentStoreResources.Requests[v1.ResourceMemory] == (resource.Quantity{}) {
+		changed = true
+		s.SegmentStoreResources.Requests[v1.ResourceMemory] = s.SegmentStoreResources.Limits[v1.ResourceMemory]
+	}
+
+	if s.SegmentStoreResources.Limits[v1.ResourceCPU] == (resource.Quantity{}) {
+		changed = true
+		s.SegmentStoreResources.Limits[v1.ResourceCPU] = resource.MustParse(DefaultSegmentStoreLimitCPU)
+	}
+
+	if s.SegmentStoreResources.Requests[v1.ResourceCPU] == (resource.Quantity{}) {
+		changed = true
+		s.SegmentStoreResources.Requests[v1.ResourceCPU] = resource.MustParse(DefaultSegmentStoreRequestCPU)
+	}
```

Comment:
I do not think this is a good idea.
If the user sets the CPU request and is not aware of the default limit, we can create a situation where the limit is below the request.

Reply:
We should always ask for the memory `limit`, while the `request` may be optional. In case of setting only `limit`, `request` can be set to `limit`, as this seems to be the behavior of Kubernetes: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

Reply:
@RaulGracia What about CPU?

Reply:
While I think that CPU is less of a problem, it may be good also bound the CPU resources similarly to what is being done here for memory. In this case, we should i) enforce that at least `limit` on CPU is set (to prevent a busy or malfunctioning pod starving all other pods in the node), ii) if `requests` is not set, default it to `limit`, and iii) if `requests` is set, check that it is <= `limit`.

Reply:
@jkhalack actually I will be modifying the `withDefaults()` method as shown below  so as to have the same check what we have for memory in addition to the checks what @RaulGracia mentioned  :
```
if s.SegmentStoreResources.Requests[v1.ResourceCPU] == (resource.Quantity{}) {
		changed = true
		s.SegmentStoreResources.Requests[v1.ResourceCPU] = s.SegmentStoreResources.Limits[v1.ResourceCPU]
	}
```